### PR TITLE
Add support for neovim's built-in LSP

### PR DESCRIPTION
**Pull request by woodyZootopia**
Fixes: #75

### CODE REVIEW

**Comment by jsdalton** [comment]
Thanks for this! I copied and pasted it into my vim config and I'm blessed with sane looking diagnostic messages. 🙇 

👍  to include this

**Comment by arthurk** [comment]
looks great! here's a screenshot with the lsp warnings and hover menu

Screen Shot 2020-12-01 at 1 35 08 PM

**Comment by woodyZootopia** [comment]
@cocopon It's been nearly one year since this PR was created, and it seems to be working fine in other people's environments too. I think it's ok to merge this PR to the master branch.

**Comment by dinhhuy258** [comment]
Maybe he already forgets about this project ><

**Comment by mikepqr** [comment]
This is a huge improvement over the current behavior, I'm using it now very happily, and I don't want to argue against merging it.

But I would like to note that this PR makes a fairly opinionated choice to use the same color for hint, warning and information.

The error groups look great in iceberg's red, but I would personally prefer hint, warning and information to not be the same color. This would be more consistent with neovim's defaults. Hint in a grey, warning in an orange and information in a blue would be consistent with neovim defaults. But I don't feel strongly about what colors those diagnostic levels should actually be. My point is that they should be different.

The four levels have meaning, and flattening three of them into a single color loses that. I also think is generally a good idea to make a colorscheme consistent with what neovim/plugin developers expect, unless it defeats the aesthetic goals of the colorscheme completely (which is not the case here IMO).

**Comment by woodyZootopia** [comment]
@mikepqr Thank you for your opinion. I agree with that.
I added another experimental commit, which uses different colors for warning/info/hint. The coloring is just an experimental one so please (anyone) tell me frankly if there is any issues with that.

**Comment by mikepqr** [comment]
Thanks! Commented over on #75.

File: src/iceberg.vim [hunk 1]
```diff
@@ -566,6 +566,75 @@ function! s:create_colors(palette) abort
...
+  endfor
+  " Info color is hard to read within floating window, so use normal text
+  " color instead.
+  call extend(rules, pgmnt#hi#group(
+        \ 'LspDiagnosticsFloatingHint', {
+        \   'ctermbg': c.pmenu_bg,
+        \   'ctermfg': c.pmenu_fg,
+        \   'guibg': g.pmenu_bg,
+        \   'guifg': g.pmenu_fg,
+        \ }))
```
Comment (cocopon) [inline src/iceberg.vim:601]: It is deprecated and can be removed.

File: src/iceberg.vim [hunk 2]
```diff
@@ -774,7 +843,6 @@ function! s:create_links() abort
   call add(links, pgmnt#hi#link('typescriptMessage', 'Normal'))
   call add(links, pgmnt#hi#link('typescriptNull', 'Constant'))
   call add(links, pgmnt#hi#link('typescriptParens', 'Normal'))
-  " }}}
```
Comment (cocopon) [inline src/iceberg.vim:777]: Unintentional deletion?

File: src/iceberg.vim [hunk 3]
```diff
@@ -774,6 +774,38 @@ function! s:create_links() abort
   call add(links, pgmnt#hi#link('typescriptMessage', 'Normal'))
   call add(links, pgmnt#hi#link('typescriptNull', 'Constant'))
   call add(links, pgmnt#hi#link('typescriptParens', 'Normal'))
+<<<<<<< HEAD
```
Comment (cocopon) [inline src/iceberg.vim]: Could you squash a commit including invalid syntax?

**Review by cocopon — CHANGES_REQUESTED** [review]
Thank you for your contribution, and sorry for the late reply... Please check the feedback.

**Comment by woodyZootopia** [comment]
@cocopon Thank you for your careful comments. Applied them all, and rebased to the latest master.

**Review by cocopon — APPROVED** [review]

**Comment by cocopon** [comment]
Merged, thank you for your contribution!

Note: I fine-tuned some coloration:
- Hint: Used `comment_fg` instead because `linenr_fg` is not designed for the normal background.
- Info: Used `lblue` instead because it is more readable than `blue`, it has distant hue from the normal background.

**Pull request merged**